### PR TITLE
Added storage driver selection and upgraded to version 2

### PR DESCRIPTION
**Pull request by joshuacox**
does what it says on the tin, i.e. 
1. I upgraded the yml files to version 2
2. I added in the VOLUME_DRIVER so I can utilize rancher-nfs
3. gave a host affinity label so I influence what set of nodes I want it on

### CODE REVIEW

File: templates/zookeeper/3/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+      io.rancher.container.hostname_override: container_name
+      io.rancher.scheduler.affinity:host_label: ${host_label}
+      io.rancher.container.start_once: true
+    environment:
+      - SERVICE_UID=10002
+      - SERVICE_GID=10002
+      - SERVICE_VOLUME=${zk_data_dir}
+    volumes:
+      - zkdata:${zk_data_dir}
+    volume_driver: local
```
Comment (rawmind0) [inline templates/zookeeper/3/docker-compose.yml]: Wrong line...
```
-    volume_driver: local
```

File: templates/zookeeper/3/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    image: rawmind/alpine-volume:0.0.2-1
+volumes:
+  zkconfig:
+    driver: ${VOLUME_DRIVER}
```
Comment (rawmind0) [inline templates/zookeeper/3/docker-compose.yml]: Need a volume per container. Every zookeeper instance need its own volumes.
```
    driver: ${VOLUME_DRIVER}
+    per_container: true
  zkdata:
    driver: ${VOLUME_DRIVER}
+    per_container: true
```

File: templates/zookeeper/3/rancher-compose.yml [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+      label: "Zk Interval (s):"
+      required: true
+      default: 60
+      type: "int"
+    - variable: host_label
+      label: "Host with Label to put zookeeper on"
+      description: |
+        Host label to use as zookeeper 'value' tag.
+        Example: 'zookeeper=true'
+      required: true
```
Comment (rawmind0) [inline templates/zookeeper/3/rancher-compose.yml]: Please make host label optional....
```
-      required: true
+      required: false
-      default: "private=true"
+      default: ""
```

File: templates/zookeeper/3/docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+      - JVMFLAGS=-Xmx${zk_mem}m -Xms${zk_mem}m
+      - ZK_DATA_DIR=${zk_data_dir}
+      - ZK_INIT_LIMIT=${zk_init_limit}
+      - ZK_MAX_CLIENT_CXNS=${zk_max_client_cxns}
+      - ZK_SYNC_LIMIT=${zk_sync_limit}
+      - ZK_TICK_TIME=${zk_tick_time}
+    labels:
+      io.rancher.scheduler.affinity:container_label_soft_ne: io.rancher.stack_service.name=$${stack_name}/$${service_name}
+      io.rancher.container.hostname_override: container_name
+      io.rancher.scheduler.affinity:host_label: ${host_label}
```
Comment (rawmind0) [inline templates/zookeeper/3/docker-compose.yml]: Optional host_label...
```
-      io.rancher.scheduler.affinity:host_label: ${host_label}
+    {{- if ne .Values.host_label ""}}
+      io.rancher.scheduler.affinity:host_label: ${host_label}
+    {{- end}}
```

File: templates/zookeeper/3/docker-compose.yml [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+      io.rancher.sidekicks: zk-volume, zk-conf
+  zk-conf:
+    labels:
+      io.rancher.scheduler.affinity:container_label_soft_ne: io.rancher.stack_service.name=$${stack_name}/$${service_name}
+      io.rancher.container.hostname_override: container_name
+      io.rancher.scheduler.affinity:host_label: ${host_label}
```
Comment (rawmind0) [inline templates/zookeeper/3/docker-compose.yml]: Optional host_label...
```
-      io.rancher.scheduler.affinity:host_label: ${host_label}
+    {{- if ne .Values.host_label ""}}
+      io.rancher.scheduler.affinity:host_label: ${host_label}
+    {{- end}}
```

File: templates/zookeeper/3/docker-compose.yml [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+      io.rancher.container.start_once: true
+    image: rawmind/rancher-zk:3.4.9
+    volumes:
+      - zkconfig:/opt/tools
+  zk-volume:
+    labels:
+      io.rancher.scheduler.affinity:container_label_soft_ne: io.rancher.stack_service.name=$${stack_name}/$${service_name}
+      io.rancher.container.hostname_override: container_name
+      io.rancher.scheduler.affinity:host_label: ${host_label}
```
Comment (rawmind0) [inline templates/zookeeper/3/docker-compose.yml]: Optional host_label...
```
-      io.rancher.scheduler.affinity:host_label: ${host_label}
+    {{- if ne .Values.host_label ""}}
+      io.rancher.scheduler.affinity:host_label: ${host_label}
+    {{- end}}
```

**Comment by rawmind0** [comment]
Hi @joshuacox ... thanks so much for the PR....

Please, take a look at the change request. There is some errors with volumes. Also, could you use go templating in docker-compose.yml.tpl to make host label optional? 

Thanks...

**Comment by joshuacox** [comment]
I think that does it, I appreciate the feedback

**Comment by rawmind0** [comment]
Cool @joshuacox, thanks... :)

LGTM...

**Pull request merged**